users/validators.py

```
@@ -16,3 +17,147 @@ class ASCIIUsernameValidator(validators.RegexValidator):
+
+    if not links and "<a" in value.lower():
+        raise ValidationError(
+            "All <a> tags must have an href attribute with proper quotes."
+        )
+
+    # Validate each URL in href attributes
+    for url in links:
+        if not is_valid_url(url):
+            raise ValidationError(
+                f"Invalid or external URL detected: {url}. Only internal links (starting with / or #) are allowed."
+            )
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)